src/org/labkey/targetedms/passport/PassportController.java

```diff
@@ -0,0 +1,395 @@
+            if(begin == null || end == null)
+                return null;
+            f.setStartIndex(begin);
+            f.setEndIndex(end);
+        } else {
+            Integer loc =Integer.parseInt(((Element) location.getElementsByTagName("position").item(0)).getAttribute("position"));
+            if(loc == null)
+                return null;
+            f.setStartIndex(loc);
+            f.setEndIndex(loc);
+        }
+        return f;
+    }
+    // returns null if no protein found
+    private IProtein getProtein(String accession)
+    {
+        SQLFragment targetedMSProteinQuery = new SQLFragment();
+        targetedMSProteinQuery.append("SELECT ps.seqid as seqid, ps.bestgenename, ps.description, ps.protsequence, ps.length, " +
+                "pg.id as pgid, pg.species, pg.preferredname, pg.runid, pg.label, " +
+                "r.dataid, r.filename, r.created, r.modified, r.formatversion " +
+                "FROM targetedms.peptidegroup pg, targetedms.runs r, prot.sequences ps " +
+                "WHERE r.id = pg.runid AND r.container = ? AND ps.seqid = pg.sequenceid " +
+                "AND pg.accession = ? LIMIT 1;");
+        targetedMSProteinQuery.add(getContainer().getId());
+        targetedMSProteinQuery.add(accession);
+        UserSchema schema = QueryService.get().getUserSchema(getUser(), getContainer(), "targetedms");
+        try {
+            Map<String, Object> map = new SqlSelector(schema.getDbSchema(), targetedMSProteinQuery).getMap();
+
+            IProtein p = new IProtein();
+            if(map == null)
+                return null;
+            p.setLabel((String) map.get("label"));
+            p.setGene((String) map.get("bestgenename"));
+            p.setSpecies((String) map.get("species"));
+            p.setPreferredname((String) map.get("preferredname"));
+            p.setPepGroupId((Integer) map.get("pgid"));
+            p.setSequenceId((Integer) map.get("seqid"));
+            p.setDescription((String) map.get("description"));
+            p.setSequence((String) map.get("protsequence"));
+            p.setLength((Integer) map.get("length"));
+
+            IFile f = new IFile();
+            f.setFileName((String) map.get("filename"));
+            f.setSoftwareVersion((String) map.get("softwareversion"));
+            f.setCreatedDate((Date) map.get("created"));
+            f.setModifiedDate((Date) map.get("modified"));
+            f.setRunId((Integer) map.get("runid"));
+
+            p.setFile(f);
+            p.setAccession(accession);
+            populateProteinKeywords(p);
+            populateUniprotData(p);
+            populatePeptides(p);
+//            populateProjects(p); TODO: project stuff needs more work - will happen next update
+            return p;
+        }
+        catch(Exception e) {
+            return null;
+        }
+
+    }
+
+    private void populateProjects(IProtein p)
```

Comment:
Let's delete this - looks to be unused and there are multiple problems in the method